crates/containerd-shim-wasm/src/sandbox/containerd/client.rs

```diff
@@ -406,32 +407,34 @@ impl Client {
             .iter()
             .filter(|x| is_wasm_layer(x.media_type(), T::supported_layers_types()));
 
-        let mut layers = vec![];
+        let mut all_layers = HashMap::new();
+        let media_type_label = precompile_label(T::name(), "media-type");
```

Comment:
Where is this `"media-type"` string coming from or how was it determined?

Reply:
Each `PrecompiledLayer` returned from `precompile` specifies the media-type of the new layer's bytes so that later when reading each child wasm layer we can synthesize a `Descriptor`.